pkg/dataobj/internal/dataset/reader.go

```diff
@@ -66,16 +66,12 @@ func NewReader(opts ReaderOptions) *Reader {
 // returns the number of rows read and any error encountered. At the end of the
 // Dataset, Read returns 0, [io.EOF].
 func (r *Reader) Read(ctx context.Context, s []Row) (n int, err error) {
+	stats := StatsFromContext(ctx)
+	stats.AddReadCalls(1)
+
 	if len(s) == 0 {
 		return 0, nil
 	}
-	// Init stats object and use the context, otherwise we create a new one every time we increment a stat.
-	var statistics *stats.Context
-	if stats.IsPresent(ctx) {
```

Comment:
I believe this was the only caller of IsPresent, so you can remove that method entirely if its unused now
I added it a couple of months ago because otherwise `stats.FromContext` will allocate every time you want to fetch the statistics object, which added up when calling Read without one